xbmc/playlists/PlayListXSPF.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ *  Copyright (C) 2018 Tyler Szabo
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PlayListXSPF.h"
+
+#include "utils/log.h"
+#include "utils/URIUtils.h"
+#include "utils/XBMCTinyXML.h"
+#include "URL.h"
+
+using namespace PLAYLIST;
+
+/*
+
+Specification: http://www.xspf.org/specs/
+
+Example:
+
+<?xml version="1.0" encoding="UTF-8"?>
+<playlist xmlns="http://xspf.org/ns/0/" version="1">
+  <title>KEXP</title>
+  <trackList>
+    <track>
+      <location>http://live-aacplus-64.kexp.org/kexp64.aac</location>
+      <title>KEXP.ORG 90.3FM - where the music matters</title>
+      <info>https://kexp.org/</info>
+      <image>https://www.kexp.org/static/assets/img/logo-white.svg</image>
+    </track>
+    <track><location>Relative/Path/To.mp3</location></track>
+    <track><location>file:///music/song_1.ogg</location></track>
+    <track><location>file:///music/song_2.flac</location></track>
+    <track><location>file:///music/song_3.mp3</location></track>
+  </trackList>
+</playlist>
+*/
+
+static constexpr char const * LOCATION_TAGNAME = "location";
```

Comment:
This diverges from our coding style... The pointer modifies the following ID, so I'd prefer `char const *LOCATION_TAGNAME`

Reply:
Thanks for clarifying. May I put this in `/docs/CODE_GUIDELINES.md`?

Reply:
There are a few more things missing in that document - I'll take it from here